demo_modules/gears/gears.js

```diff
@@ -286,15 +288,35 @@ class GearsClient extends ModuleInterface.Client {
       const baseRadius = baseDiameter / 2;
       const outsideRadius = pitchRadius + addendum;
       const rootRadius = outsideRadius - wholeDepth;
-      
+
+      this.gearDetails_[key] = {
+        pitchDiameter: pitchDiameter,
+        diametralPitch: diametralPitch,
+        addendum: addendum,
+        wholeDepth: wholeDepth,
+        radiusAngle: radiusAngle,
+        baseDiameter: baseDiameter,
+        baseRadius: baseRadius,
+        outsideRadius: outsideRadius,
+        rootRadius: rootRadius
+      };
+    }
+    return this.gearDetails_[key];
+  }
+  getGearPath_(holes, pitchRadius, numberOfTeeth) {
+    // Rather than always making a new gear path, consult our cache.
+    const key = [holes, pitchRadius, numberOfTeeth].join(',');
+    if (!this.gearPaths_[key]) {
+      const details = this.getGearDetails_(pitchRadius, numberOfTeeth);
```

Comment:
If you *want*, you can write const {rootRadius, baseRadius, ...} = this.getGearDetails_

which avoids needing to add details. in front of all of the use sites.